cfn-resources/cluster/cmd/resource/resource.go

```diff
@@ -197,6 +201,11 @@ func Update(req handler.Request, prevModel *Model, currentModel *Model) (handler
 		return updateClusterCallback(client, currentModel, *currentModel.ProjectId)
 	}
 
+	pe := currentModel.validateDefaultLabel()
+	if pe != nil {
+		return *pe, nil
+	}
+
```

Comment:
if the cluster has been created with CFN, it will have the label and then `validateDefaultLabel` will fail. I believe we want to add the label only for cluster creation not update @Zuhairahmed to confirm

Reply:
let me try that, because if its coming from the current model, the user should not be providing the tag, ill test that just in case

Reply:
so this is the case, if the user does not provide the label in the template used as an input on the update is fine, if the user provides the label as the update imput, it will brake and the user will need to provide a new template without the label

Reply:
@andreaangiolillo is correct. @Adelmar92 if not possible to improve further can you make ^^ this clear in Readme file for this resource.

Reply:
> so this is the case, if the user does not provide the label in the template used as an input on the update is fine, if the user provides the label as the update imput, it will brake and the user will need to provide a new template without the label

isn't it because of `validateDefaultLabel`? This function returns an error if the model has a label (which is wrong).  The idea is that users can add as many labels as they want. We should add the `Infrastructure Tool` all the time that the cluster is created with CFN. As a result, ]

- if the user does not provide a label, we add `Infrastructure Tool`
- if the user provides 4 labels, we add 4 labels + `Infrastructure Tool`

Let me know if I am missing something or if it is not clear. I know that it can be confusing. Thanks!

Reply:
My understanding is that we would replace it. @Zuhairahmed to confirm here

Reply:
so we discuss this with @Zuhairahmed im modifying the behaviour like this:

- if the user provides the label Infrastructure Tool we are keepeng it
- if the user does not provides the label Infrastructure Tool we are adding it